src/features/query-bar/components/filter/Condition.js

```diff
@@ -150,91 +156,122 @@ const Condition = ({
         const { valueKey, valueType } = condition;
 
         const isValueSet = valueKey !== null && valueKey !== '';
-
-        const message = valueType === 'date' ? messages.tooltipSelectDateError : messages.tooltipSelectValueError;
-
-        const error = areErrorsEnabled && !isValueSet ? formatMessage(message) : null;
+        const message = (
+            <FormattedMessage
+                {...(valueType === DATE ? messages.tooltipSelectDateError : messages.tooltipSelectValueError)}
+            />
+        );
+        const error = areErrorsEnabled && !isValueSet ? message : null;
 
         return error;
     };
 
     const renderDeleteButton = () => {
         return (
-            <div className="filter-item-delete-button">
+            <div className="condition-delete-button">
                 <button className="delete-button" onClick={onDeleteButtonClick} type="button">
                     <IconClose width={deleteButtonIconWidth} height={deleteButtonIconHeight} color="#999EA4" />
                 </button>
             </div>
         );
     };
 
-    const renderPrefixField = () => {
+    const renderConnectorField = () => {
+        let message = '';
+        switch (selectedConnector) {
+            case AND:
+                message = <FormattedMessage {...messages.connectorAndText} />;
+                break;
+            case OR:
+                message = <FormattedMessage {...messages.connectorOrText} />;
+                break;
+            default:
+                break;
+        }
+
+        const connectorOptions = getFormattedOptions(
+            [AND, OR].map(connector => ({
+                displayName: connector,
+                value: connector,
+            })),
+        );
+
         return (
-            <div className="filter-item-prefix-container">
-                <p className="filter-item-prefix-text">
-                    <FormattedMessage {...messages.prefixButtonText} />
-                </p>
+            <div className="condition-connector">
+                {index === 0 ? (
+                    <p className="condition-connector-text">
+                        <FormattedMessage {...messages.connectorWhereText} />
+                    </p>
+                ) : (
+                    <SingleSelectField
+                        isDisabled={false}
+                        onChange={onConnectorChange}
+                        options={connectorOptions}
+                        placeholder={message}
```

Comment:
Why use placeholder instead of selectedValue? 

See the purpose of each prop here. 

https://git.dev.box.net/Box/box-react-ui/blob/e207854a487574b7be3ec6a7cad9c8c55afac271/src/components/select-field/SingleSelectField.js

Reply:
Only selectedValue is needed, removing placeholder